apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/FuzzySearch.tsx

```diff
@@ -149,16 +124,28 @@ class FuzzySearch extends PureComponent<FuzzySearchProps, FuzzySearchState> {
                     if (this.state.cache[this.state.value]) {
                         this.setState({ results: this.state.cache[this.state.value] });
                     } else {
-                        try {
-                            const result = search({ query: this.state.value, numResults: 10 });
-                            this.setState({
-                                cache: { ...this.state.cache, [this.state.value]: result },
-                                results: result,
-                            });
-                        } catch (e) {
-                            this.setState({ results: {} });
-                            console.error(e);
-                        }
+                        this.setState({ results: {}, loading: true }, () => {
+                            window.clearTimeout(this.state.pendingRequest);
+                            const pendingRequest = window.setTimeout(
+                                () =>
+                                    trpc.search.doSearch
```

Comment:
In theory, couldn't there be a race condition where the older result arrives later and mutates the search results out of turn? Maybe you should look into aborting the call or otherwise invalidating the state update.